### PR TITLE
12: update test isolation docs to use true/false instead of on/off

### PR DESCRIPTION
**Pull request by emilyrohrbough**
update test isolation docs & apply product review suggestions

### CODE REVIEW

File: content/guides/core-concepts/test-isolation.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+It is important to note that disabling test isolation may improve the overall
+performance of end-to-end tests, however, it can also cause state to "leak"
+between tests. This can make later tests dependent on the results of earlier
+tests, and potentially cause misleading test failures. It is important to be
+extremely mindful of how tests are written when using this mode, and ensure that
+tests continue to run independently of one another.
+
+The best way to ensure your tests are independent is to add a `.only()` to your
+test and verify it can run successfully without the test before it.
```
Comment (emilyrohrbough) [inline content/guides/core-concepts/test-isolation.md:124]: likely need to add an example here -- same as migration guide?

File: content/guides/core-concepts/test-isolation.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+<Icon name="check-circle" color="green"></Icon> **Best Practice:** Tests should
+always be able to be run independently from one another **and still pass**.
+
+</Alert>
+
+As stated in our mission, we hold ourselves accountable to champion a testing
+process that actually works, and have built Cypress to guide developers towards
+writing independent tests from the start.
+
+We do this by cleaning up state _before_ each test to ensure that the operation
```
Comment (mjhenkes) [inline content/guides/core-concepts/test-isolation.md:28]:
```suggestion
We do this by cleaning up browser state _before_ each test to ensure that the operation
```

Comment (mjhenkes) [reply content/guides/core-concepts/test-isolation.md:28]: i know we clean up other state too, but it seems like browser state is the relevant state to this conversation 🤷‍♂️

File: content/guides/core-concepts/test-isolation.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+of one test does not affect another test later on. The goal for each test should
+be to **reliably pass** whether run in isolation or consecutively with other
+tests. Having tests that depend on the state of an earlier test can potentially
+cause nondeterministic test failures which makes debugging challenging.
```
Comment (mjhenkes) [inline content/guides/core-concepts/test-isolation.md:32]:
```suggestion
cause nondeterministic test failures which make debugging challenging.
```

File: content/guides/references/migration-guide.md [hunk 4]
```diff
@@ -83,8 +82,8 @@ it('validates the change', () => {
 })
 ```
 
-After migrating, when `testIsolation='on'`, this flow would need to be contained
-within a single test. While the above practice has always been
+After migrating and `testIsolation=true` by default, this flow would need to be
```
Comment (mjhenkes) [inline content/guides/references/migration-guide.md:85]: This sentence seemed awkward

```suggestion
After migrating, when `testIsolation=true` by default, this flow would need to be
```

File: content/guides/core-concepts/test-isolation.md (hunk 2 above)
Comment (mjhenkes) [reply content/guides/core-concepts/test-isolation.md:28]: eh, i think this is fine in the context of the whole article.

**Review by mjhenkes — APPROVED** [review]
looks good, a couple of minor suggestions.

File: content/api/commands/session.md [hunk 5]
```diff
@@ -823,9 +823,10 @@ generate random unique ids if an arbitrary name-space does not meet your needs.
 
 #### Why are all my Cypress commands failing after calling `cy.session()`?
 
-When test isolation is `on`, ensure that you're calling
-[`cy.visit()`](/api/commands/visit) after calling `cy.session()`, otherwise your
-tests will be running on a blank page.
+When
+[`testIsolation`](/guides/core-concepts/writing-and-organizing-tests#Test-Isolation)is
```
Comment (AtofStryker) [inline content/api/commands/session.md:827]:
```suggestion
[`testIsolation`](/guides/core-concepts/writing-and-organizing-tests#Test-Isolation) is
```

File: content/guides/core-concepts/test-isolation.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+clean browser context such that the application or component under test behaves
+consistently when ran. This behavior is described as `testIsolation`.
+
+The test isolation is a global configuration and can be overridden for
+end-to-end testing at the `describe` level with the
+[`testIsolation`](/guides/references/configuration#Global) option.
+
+## Test Isolation in End-to-End Testing
+
+Cypress supports the enabled or disabling test isolation in end-to-end testing
```
Comment (chrisbreiding) [inline content/guides/core-concepts/test-isolation.md:55]:
```suggestion
Cypress supports enabling or disabling test isolation in end-to-end testing
```

**Review by AtofStryker — APPROVED** [review]

File: content/guides/references/migration-guide.md [hunk 7]
```diff
@@ -124,27 +123,27 @@ application logins, while users also benefit from the test isolation guardrails
...
 })
 ```
 
-It is important to note that while turning test isolation `off` may improve the
-overall performance of end-to-end tests, it can however cause state to "leak"
-between tests. This can make later tests dependent on the results of earlier
-tests, and potentially cause misleading test failures. It is important to be
-extremely mindful of how tests are written when using this mode, and ensure that
-tests continue to run independently of one another.
+It is important to note that disabling test isolation may improve the overall
```
Comment (chrisbreiding) [inline content/guides/references/migration-guide.md:135]:
```suggestion
It is important to note that while disabling test isolation may improve the overall
```

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**